README.md

```diff
@@ -241,30 +241,54 @@ If you set `exposeRoute` to `true` the plugin will expose the documentation with
 | `'/documentation/*'`    | External files that you may use in `$ref`  |
 
 <a name="register.options.transform"></a>
-#### Transforms
+#### Transform
 
-To use different schemas such as [Joi](https://github.com/hapijs/joi) you can pass a synchronous `transform` method in the options to convert them back to standard JSON schemas expected by this plugin to generate the documentation (`dynamic` mode only).
+By passing a synchronous `transform` function you can modify the route's url and schema.
+
+Some possible uses of this are:
+
+- add the `hide` flag on schema according to your own logic based on url & schema
+- altering the route url into something that's more suitable for the api spec
+- using different schemas such as [Joi](https://github.com/hapijs/joi) and transforming them to standard JSON schemas expected by this plugin
+
+This option is available in `dynamic` mode only.
+
+Examples of all the possible uses mentioned:
 
 ```js
 const convert = require('joi-to-json')
 
 fastify.register(require('fastify-swagger'), {
   swagger: { ... },
-  ...
-  transform: schema => {
+  transform: ({ schema, url }) => {
     const {
       params = undefined,
       body = undefined,
       querystring = undefined,
-      ...others
+      headers = undefined,
+      response = undefined,
+      ...rest
```

Comment:
No need to assign undefined as default fallback values here, it is automatically done when using object destructuring.